generated/nidaqmx/_library_interpreter.py

```diff
@@ -0,0 +1,2227 @@
+        check_for_error(error_code)
+
+    def disable_start_trig(self, task):
+
+        cfunc = lib_importer.windll.DAQmxDisableStartTrig
+        if cfunc.argtypes is None:
+            with cfunc.arglock:
+                if cfunc.argtypes is None:
+                    cfunc.argtypes = [
+                        lib_importer.task_handle]
+
+        error_code = cfunc(
+            task)
+        check_for_error(error_code)
+
+    def disconnect_terms(self, source_terminal, destination_terminal):
+
+        cfunc = lib_importer.windll.DAQmxDisconnectTerms
+        if cfunc.argtypes is None:
+            with cfunc.arglock:
+                if cfunc.argtypes is None:
+                    cfunc.argtypes = [
+                        ctypes_byte_str, ctypes_byte_str]
+
+        error_code = cfunc(
+            source_terminal, destination_terminal)
+        check_for_error(error_code)
+
+    def export_signal(self, task, signal_id, output_terminal):
+
+        cfunc = lib_importer.windll.DAQmxExportSignal
+        if cfunc.argtypes is None:
+            with cfunc.arglock:
+                if cfunc.argtypes is None:
+                    cfunc.argtypes = [
+                        lib_importer.task_handle, ctypes.c_int,
+                        ctypes_byte_str]
+
+        error_code = cfunc(
+            task, signal_id, output_terminal)
+        check_for_error(error_code)
+
+    def register_done_event(
+            self, task, options, callback_function, callback_data):
+
+        DAQmxDoneEventCallbackPtr = ctypes.CFUNCTYPE(
+            lib_importer.task_handle, ctypes.c_int32)        
+        
+        cfunc = lib_importer.windll.DAQmxRegisterDoneEvent
+        
+        with cfunc.arglock:
+            if callback_function is not None:
+                callback_method_ptr = DAQmxDoneEventCallbackPtr(callback_function)
+                cfunc.argtypes = [
+                    lib_importer.task_handle, ctypes.c_uint,
+                        DAQmxDoneEventCallbackPtr, ctypes.c_void_p]
+            else:
+                callback_method_ptr = None
+                cfunc.argtypes = [
+                    lib_importer.task_handle, ctypes.c_uint,
+                        ctypes.c_void_p, ctypes.c_void_p]
+        
+            error_code = cfunc(
+                task, options, callback_function, callback_data)
+        check_for_error(error_code)
+
+
+    def register_every_n_samples_event(
+            self, task, every_n_samples_event_type, n_samples, options,
+            callback_function, callback_data):
+
+        DAQmxEveryNSamplesEventCallbackPtr = ctypes.CFUNCTYPE(
+            lib_importer.task_handle, ctypes.c_int32,
+                        ctypes.c_uint32)        
+        
+        cfunc = lib_importer.windll.DAQmxRegisterEveryNSamplesEvent
+        
+        with cfunc.arglock:
+            if callback_function is not None:
+                callback_method_ptr = DAQmxEveryNSamplesEventCallbackPtr(callback_function)
```

Comment:
This creates a ctypes object representing the callback function, and it is critical that this object is not garbage collected until the event is unregistered. Otherwise, when DAQmx calls the callback, it will crash. 

This is why `Task.register_every_n_samples_event` does this:
```
                self._every_n_acquired_event_callbacks.append(
                    callback_method_ptr)
```

The lists are declared here: https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/task.py#L384
```
        # These lists keep C callback objects in memory as ctypes doesn't.
        # Program will crash if callback is made after object is garbage
        # collected.
        self._done_event_callbacks = []
        self._every_n_transferred_event_callbacks = []
        self._every_n_acquired_event_callbacks = []
        self._signal_event_callbacks = []
```

I think it's okay to add these lists to the library interpreter. The gRPC interpreter won't need them, it will need something different.

FYI: another way this *could have worked* is to use a single, global callback object per function and use the callback_data to distinguish between tasks. This might have eliminated the need for these lists, assuming that you can use an existing object like the `Task` or the `LibraryInterpreter`. However, I think something like this is outside the scope of the library interpreter refactor, since the main goal is to add gRPC support and adding the lists to the library interpreter doesn't detract from this.

Reply:
Added callback lists to library interpreter